include/bluetooth/crypto.h

```diff
@@ -102,19 +102,19 @@ int bt_ccm_decrypt(const uint8_t key[16], uint8_t nonce[13], const uint8_t *enc_
  *
  *  @param key       128 bit MS byte first key
  *  @param nonce     13 byte MS byte first nonce
- *  @param enc_data  Buffer to place encrypted data in
+ *  @param plaintext Plaintext buffer to encrypt
  *  @param len       Length of the encrypted data
  *  @param aad       Additional input data
  *  @param aad_len   Additional input data length
- *  @param plaintext Plaintext buffer to encrypt
+ *  @param enc_data  Buffer to place encrypted data in
  *  @param mic_size  Size of the trailing MIC (in bytes)
  *
  *  @retval 0        Successfully encrypted the data.
  *  @retval -EINVAL  Invalid parameters.
  */
-int bt_ccm_encrypt(const uint8_t key[16], uint8_t nonce[13], const uint8_t *enc_data,
-		   size_t len, const uint8_t *aad, size_t aad_len,
-		   uint8_t *plaintext, size_t mic_size);
+int bt_ccm_encrypt(const uint8_t key[16], uint8_t nonce[13],
+		   const uint8_t *plaintext, size_t len, const uint8_t *aad,
+		   size_t aad_len, uint8_t *enc_data, size_t mic_size);
```

Comment:
This still isn't consistent with the parameter names of actual implementation however:
```
int bt_ccm_encrypt(const uint8_t key[16], uint8_t nonce[13], const uint8_t *msg,
		   size_t msg_len, const uint8_t *aad, size_t aad_len,
		   uint8_t *out_msg, size_t mic_size)
```
Could you make sure the parameter names are aligned between the header file and the c-file?

Reply:
`bt_ccm_decrypt` has the same issue

Reply:
Indeed. Both should be fixed to have the same parameter naming in the c- and header-files.

Reply:
Is there a way we can verify this automatically in CI? e.g. using libclang or other existing linters.
This specific function is especially nasty because the data types did still match.

Reply:
@jhedberg I did that in a separate commit now, I didn't know if a rebase is accepted here (or maybe even preferred?)

Reply:
@BeckmaR thanks, however please squash the changes into a single commit and `push --force` to the PR branch. This is the preferred way with Zephyr development (i.e. instead of adding commits to fix something in an earlier commit in the same PR just do the fix into the original instead). You can also rebase to latest master in the same go, but this makes less of a difference since CI anyway rebases before attempting to build and merging the PR will also rebase on top of the latest master.

Reply:
Done

Reply:
> Is there a way we can verify this automatically in CI? e.g. using libclang or other existing linters.
> This specific function is especially nasty because the data types did still match.

@M1cha that would indeed be nice if CI could catch it, however apparently something like that is not enabled for the current checks. I'm sure a contribution to add such a check would be welcome.